Etabs_Adapter/CRUD/Create/Bar.cs

```diff
@@ -153,10 +153,15 @@ private bool SetObject(Bar bhBar)
                 offset2[2] = offset.End.Y;
             }
 
-            if (m_model.FrameObj.SetInsertionPoint(name, (int)bhBar.InsertionPoint(), false, bhBar.ModifyStiffnessInsertionPoint(), ref offset1, ref offset2) != 0)
-            {
-                CreatePropertyWarning("Insertion point and perpendicular offset", "Bar", name);
-                ret++;
+
+            // Avoid following operation if ETABS Version is ETABS21...
+            if (!this.etabsVersion.Contains("21")) {
```

Comment:
Would try to make this a bit more safe, to avoid a null case, as well as the off chance that the build or the patch could be 21. Would suggest something like the following:

```suggestion
            // Avoid following operation if ETABS Version is ETABS21...
            string majorVersion = "";
            if(this.etabsVersion != null && this.etabsVersion.Contains("."))
                majorVersion = this.etabsVersion.Split('.')[0];

            if (majorVersion != "21") 
            {
```

As an alternative to the above, you could do the work of finding the major version when you get it out in the first place in the adapter constructor. I will leave it to you to decide what you like best.